### PR TITLE
Move climate setting from objective to scenario settings window

### PR DESCRIPTION
**Pull request by Broxzier**
This implements #8222.

Note that this does not work over multiplayer, it simply sets the climate on the client, just like the console command currently does. Now that players have easier access, this issue may become more apparent.

### CODE REVIEW

**Comment by ZehMatt** [comment]
Would adding a game action to change the climate actually do the trick?

**Comment by Gymnasiast** [comment]
I'd prefer if you made it network safe. After all, desyncs are very frequently mentioned as dampening the multiplayer experience - and this one we know the cause of, which should make it easy to address.

**Comment by Broxzier** [comment]
Sure, I was planning on splitting that part into its own PR, but I can create it for this one if you prefer.

**Comment by Gymnasiast** [comment]
Ah, good to hear. I don't mind that much, as long as the network bits follow soon.

File: src/openrct2/Game.h [hunk 1]
```diff
@@ -89,7 +89,8 @@ enum GAME_COMMAND
     GAME_COMMAND_MODIFY_TILE,
     GAME_COMMAND_EDIT_SCENARIO_OPTIONS,
     GAME_COMMAND_PLACE_PEEP_SPAWN, // GA, TODO: refactor to separate array for just game actions
-    GAME_COMMAND_COUNT
+    GAME_COMMAND_SET_CLIMATE,
```
Comment (Broxzier) [inline src/openrct2/Game.h]: TODO: add `GA` comment (waiting for CI to turn (hopefully) turn green first.
```suggestion
    GAME_COMMAND_SET_CLIMATE, // GA
```

File: src/openrct2-ui/windows/EditorScenarioOptions.cpp [hunk 2]
```diff
@@ -510,6 +532,24 @@ static void window_editor_scenario_options_financial_resize(rct_window* w)
...
 }
 
+static void window_editor_scenario_options_show_climate_dropdown(rct_window* w)
+{
+    int32_t i;
+    rct_widget* dropdownWidget;
+
+    dropdownWidget = &w->widgets[WIDX_CLIMATE];
+
+    for (i = 0; i < 4; i++)
```
Comment (Broxzier) [inline src/openrct2-ui/windows/EditorScenarioOptions.cpp]:
```suggestion
    for (i = 0; i < CLIMATE_COUNT; i++)
```

File: src/openrct2-ui/windows/EditorScenarioOptions.cpp [hunk 3]
```diff
@@ -510,6 +532,24 @@ static void window_editor_scenario_options_financial_resize(rct_window* w)
...
+    {
+        gDropdownItemsFormat[i] = STR_DROPDOWN_MENU_LABEL;
+        gDropdownItemsArgs[i] = ClimateNames[i];
+    }
+    window_dropdown_show_text_custom_width(
+        w->x + dropdownWidget->left, w->y + dropdownWidget->top, dropdownWidget->bottom - dropdownWidget->top + 1,
+        w->colours[1], 0, DROPDOWN_FLAG_STAY_OPEN, 4, dropdownWidget->right - dropdownWidget->left - 3);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/EditorScenarioOptions.cpp]:
```suggestion
        w->colours[1], 0, DROPDOWN_FLAG_STAY_OPEN, CLIMATE_COUNT, dropdownWidget->right - dropdownWidget->left - 3);
```

**Comment by Gymnasiast** [comment]
Looking good codewise (assuming you'll merge your suggestions). Haven't tested it, though.

It's probably also nice as an example on how to add a GameAction - not too much code, so easy to oversee.

**Pull request merged**